flow/layers/image_filter_layer.cc

```diff
@@ -83,23 +85,30 @@ void ImageFilterLayer::Paint(PaintContext& context) const {
   FML_DCHECK(needs_painting(context));
 
   AutoCachePaint cache_paint(context);
-
-  if (layer_raster_cache_item_->IsCacheChildren()) {
-    cache_paint.setImageFilter(transformed_filter_);
-  }
-  if (layer_raster_cache_item_->Draw(context, cache_paint.sk_paint())) {
-    return;
+  if (context.raster_cache) {
+    if (layer_raster_cache_item_->IsCacheChildren()) {
+      cache_paint.setImageFilter(transformed_filter_.get());
+    }
+    if (layer_raster_cache_item_->Draw(context, cache_paint.sk_paint())) {
+      return;
+    }
   }
 
-  cache_paint.setImageFilter(filter_);
-
-  // Normally a save_layer is sized to the current layer bounds, but in this
-  // case the bounds of the child may not be the same as the filtered version
-  // so we use the bounds of the child container which do not include any
-  // modifications that the filter might apply.
-  Layer::AutoSaveLayer save_layer = Layer::AutoSaveLayer::Create(
-      context, child_paint_bounds(), cache_paint.sk_paint());
-  PaintChildren(context);
+  cache_paint.setImageFilter(filter_ ? filter_.get() : nullptr);
+  if (context.leaf_nodes_builder) {
+    context.leaf_nodes_builder->saveLayer(&child_paint_bounds(),
+                                          cache_paint.dl_paint());
+    PaintChildren(context);
+    context.leaf_nodes_builder->restore();
```

Comment:
(At some point we should probably create a new DL-based AutoSave object?)

Reply:
AutoSaveLayer support DisplayListBuilder